### PR TITLE
fixing 5375 - update buildable fields crash

### LINKED ISSUE #5375: Segfault in `DefaultAI::update_all_buildable_fields`
The game always hangs-up at the same moment. It simply stops working, at 0% CPU and 365 MB RAM usage. When I wait some seconds and then click somewhere, Windows shows a small window: "Widelands - realtime strategy game funktioniert nicht mehr" (eng: "... has stopped working"). The task gets closed when hitting the button to close the program.

1. Run Widelands
2. Load save game [wl_autosave_00.zip](https://github.com/widelands/widelands/files/8556060/wl_autosave_00.zip)
4. wait ~1 minute

Widelands shoudn't hang-up.

screenshot

 - OS: Windows 7 Pro 64 bit
 - Widelands 1.1~git25728[efd890f@HEAD] (Release)

### PR DESCRIPTION
**Pull request by tibor95**
Calling front() on empty deque... caused by wrongly constructed while and unsigned int underflow

### CODE REVIEW

File: src/ai/defaultai.cc [hunk 1]
```diff
@@ -1364,18 +1364,21 @@ void DefaultAI::update_all_buildable_fields(const Time& gametime) {
...
 
-	verb_log_dbg_time(
-	   gametime,
-	   " first round: %2d fields updated. Fields unupdated: Spec: %d, Mid: %d, Big: %d. Invalid "
-	   "fields found: %3d\n",
-	   updated_fields_count, special_fields_to_prefer[kSpecialFieldPos],
-	   special_fields_to_prefer[kMediumlFieldPos], special_fields_to_prefer[kBigFieldPos],
-	   invalidated_bf_count);
+	verb_log_dbg_time(gametime,
+	                  " first round: %2d of %3lu fields updated. Fields unupdated: Spec: %d, Mid: "
```
Comment (hessenfarmer) [inline src/ai/defaultai.cc]:
```suggestion
	                  " first round: %2d of %3llu fields updated. Fields unupdated: Spec: %d, Mid: "
```
my build environment complains this (buildable_fields.size())to be long long unsigned int

**Review by hessenfarmer — APPROVED** [review]
fix confirmed.  
one small nit from my compiler.  
rest LGTM

**Comment by tibor95** [comment]
Here is example how this should be done:
https://github.com/widelands/widelands/blob/5c62ef17e13583556752ff2481148e0d90ad57e1/src/ai/defaultai.cc#L526

**Comment by tibor95** [comment]
I am not home and cannot make a change with that PRIuS now, and cannot find an way via WEB how to propose it...

**Comment by hessenfarmer** [comment]
did it by web. you need to go to the branch and open the file you want to edit then click on the pencil in the upper right corner.

**Comment by tibor95** [comment]
all passed

**Comment by hessenfarmer** [comment]
thanks for the fix

**Pull request merged**